primitives/src/sentry.rs

```diff
@@ -1,16 +1,17 @@
-use crate::validator::{ApproveState, Heartbeat, MessageTypes, NewState};
+use crate::validator::{Heartbeat, MessageTypes};
 use crate::{BigNum, Channel};
 use chrono::{DateTime, Utc};
 use serde::{Deserialize, Serialize};
+use serde_hex::{SerHex, StrictPfx};
 use std::collections::HashMap;
 
 #[derive(Serialize, Deserialize, Debug)]
 #[serde(rename_all = "camelCase")]
 pub struct LastApproved {
     /// NewState can be None if the channel is brand new
-    pub new_state: Option<NewState>,
+    pub new_state: Option<ValidatorMessage>,
```

Comment:
This forces us to always check if it's actually a `NewState`, isn't it better to leave the explicit types here?

Reply:
What's returned from sentry isn't `NewState` but a `ValidatorMessage`

Reply:
but we can move this match where we fetch it, instead of doing it everywhere else we use `LastApproved`

Reply:
Right now, we only have a single place that we use `LastApproved`, I'd say It's okay right now but if it occurs that we use it at several places with this same requirements then we could abstract it

Reply:
It's just being less type safe on the top level as it's not bound to what it is actually meaning. I.e. the field says `new_state`, but the value is `ValidatorMessage`, which might not be a `ValidatorMessage::NewState`